### ISSUE TITLE
liquidjs Regression on Relative Includes

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#1995]
Related to https://github.com/harttle/liquidjs/issues/395#issuecomment-933007299

Locked to `liquidjs` 9.25.x starting in Beta 1.

cc @paulrobertlloyd

**Comment by zachleat** [comment]
Shipping with 1.0.0-beta.5

**Comment by paulrobertlloyd** [comment]
Hmmm… not sure if this is a bug, or a change in the intended behaviour. I want to use `setLiquidOptions` so that I can add some `globals`:

```js
eleventyConfig.setLiquidOptions({
  globals: {
    app: require('./src/data/app.js'),
    dates: require('./src/data/dates.js'),
    navigation: require('./src/data/navigation.js'),
    srcsets: require('./src/data/srcsets.js'),
  },
});

…

return {
  dir: {
    input: './src',
    output: 'www',
    includes: 'includes',
    layouts: 'layouts',
    data: 'data',
  },
  templateFormats: ['liquid', 'md'],
};
```

However I get the following error:

```bash
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] > Having trouble writing template: www/bookmarks/2010/08/facebook_places/index.html

`TemplateWriterWriteError` was thrown
[11ty] > undefined is not iterable (cannot read property Symbol(Symbol.iterator)), file:./src/layouts/bookmark.liquid, line:1, col:1

`RenderError` was thrown
[11ty] > undefined is not iterable (cannot read property Symbol(Symbol.iterator))

`TypeError` was thrown:
[11ty]     TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
        at __values (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:91:46)
        at Render.<anonymous> (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:1101:35)
        at step (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:84:23)
        at Object.next (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:65:53)
        at reduce (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:1021:25)
        at toThenable (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:1016:16)
        at reduce (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:1028:16)
        at Object.then (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:989:43)
        at reduce (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:1028:40)
        at Object.then (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:989:43)
```

If I add `layouts` and `partials` to `setLiquidOptions`, my site builds fine:

```js
eleventy.setLiquidOptions({
  cache: true,
  globals: {
    app: require('./src/data/app.js'),
    dates: require('./src/data/dates.js'),
    navigation: require('./src/data/navigation.js'),
    srcsets: require('./src/data/srcsets.js'),
  },
  layouts: './src/layouts',
  partials: './src/includes',
});
```

Does `setLiquidOptions` mean any values for the `dir.includes` and `dir.layouts` options get ignored? Is that the intended behaviour?

**Comment by zachleat** [comment]
@paulrobertlloyd was this resolved with the 1.0 stable release? Seems to work okay for me with this config file:

```js
module.exports = function(eleventyConfig) {
  eleventyConfig.setLiquidOptions({
    globals: {
      app: { hi: 123 },
      dates: {},
      navigation: {},
      srcsets: {},
    },
  });

  return {
    dir: {
      input: '.',
      output: 'www',
      includes: 'includes',
      layouts: 'layouts',
      data: 'data',
    },
    templateFormats: ['liquid', 'md'],
  };
};
```

**Comment by paulrobertlloyd** [comment]
Afraid not. With the following config:

```js
module.exports = function (eleventy) {
  eleventy.setLiquidOptions({
    cache: true,
    globals: {
      app: require('./src/data/app.js'),
      dates: require('./src/data/dates.js'),
      navigation: require('./src/data/navigation.js'),
    },
  });

  return {
    dir: {
      input: 'src',
      output: 'www',
      includes: 'includes',
      layouts: 'layouts',
      data: 'data',
    },
    templateFormats: ['liquid', 'md', '11ty.js'],
  };
};
```

I get the following error:

```sh
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] 1. Having trouble writing template: "www/notes/1493239140/index.html" (via EleventyTemplateError)
[11ty] 2. ENOENT: Failed to lookup "default" in "src/includes,src", file:./src/layouts/note.liquid, line:1, col:1 (via RenderError)
[11ty] 3. ENOENT: Failed to lookup "default" in "src/includes,src" (via Error)
[11ty] 
[11ty] Original error stack trace: Error: ENOENT: Failed to lookup "default" in "src/includes,src"
[11ty]     at Loader.lookupError (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:974:19)
[11ty]     at Loader.<anonymous> (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:887:37)
[11ty]     at step (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:87:23)
[11ty]     at Object.next (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:68:53)
[11ty]     at reduce (/Users/paulrobertlloyd/Sites/paulrobertlloyd-v4/node_modules/liquidjs/dist/liquid.node.cjs.js:1044:25)
[11ty] Copied 1504 files / Wrote 0 files in 4.54 seconds (v1.0.1-canary.3)
```

It seems to be looking for the `default` layout in the includes folder, thus ignoring the value given for `dir.layouts`.

Explicitly adding values for `partials` and `layouts` to `setLiquidOptions` fixes this issue:

```js
eleventy.setLiquidOptions({
  cache: true,
  globals: {
    app: require('./src/data/app.js'),
    dates: require('./src/data/dates.js'),
    navigation: require('./src/data/navigation.js'),
  },
  layouts: './src/layouts',
  partials: './src/includes',
});
```
